workspaces/ballerina/ballerina-side-panel/src/components/editors/ExpandedEditor/modes/PromptMode.tsx

```diff
@@ -0,0 +1,185 @@
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import React from "react";
+import styled from "@emotion/styled";
+import { ThemeColors } from "@wso2/ui-toolkit";
+import { EditorModeWithPreviewProps } from "./types";
+import { MarkdownToolbar } from "../controls/MarkdownToolbar";
+import { MarkdownPreview } from "../controls/MarkdownPreview";
+
+const TextArea = styled.textarea`
+    width: 100%;
+    min-height: 500px;
+    padding: 12px !important;
+    fontSize: 13px;
+    font-family: var(--vscode-editor-font-family);
+    background: var(--input-background);
+    color: ${ThemeColors.ON_SURFACE};
+    border: 1px solid ${ThemeColors.OUTLINE_VARIANT};
+    border-radius: 0 0 4px 4px;
+    border-top: none;
+    resize: vertical;
+    outline: none;
+    box-sizing: border-box;
+`;
+
+const TEXTAREA_ID = "prompt-textarea";
+
+/**
+ * Prompt mode editor - textarea with markdown toolbar and preview support
+ */
+export const PromptMode: React.FC<EditorModeWithPreviewProps> = ({
+    value,
+    onChange,
+    isPreviewMode,
+    onTogglePreview,
+    field
+}) => {
+    /**
+     * Handles Enter key to automatically continue lists (similar to GitHub comments)
+     */
+    const handleKeyDown = (e: React.KeyboardEvent<HTMLTextAreaElement>) => {
+        if (e.key !== 'Enter' || e.shiftKey || e.ctrlKey || e.metaKey || e.altKey) {
+            return;
+        }
+
+        const textarea = e.currentTarget;
+        const cursorPosition = textarea.selectionStart;
+        const textBeforeCursor = value.substring(0, cursorPosition);
+        const textAfterCursor = value.substring(cursorPosition);
+
+        // Find the start of the current line
+        const lastNewlineIndex = textBeforeCursor.lastIndexOf('\n');
+        const currentLine = textBeforeCursor.substring(lastNewlineIndex + 1);
+
+        // Check for unordered list (- or *)
+        const unorderedMatch = currentLine.match(/^(\s*)([-*])\s+(.*)$/);
+        if (unorderedMatch) {
+            const [, indent, marker, content] = unorderedMatch;
+
+            // If the list item is empty (just the marker), remove it and exit list mode
+            if (!content.trim()) {
+                e.preventDefault();
+                const newValue = textBeforeCursor.substring(0, lastNewlineIndex + 1) + '\n' + textAfterCursor;
+                onChange(newValue);
+                // Set cursor position after both newlines
+                setTimeout(() => {
+                    textarea.selectionStart = textarea.selectionEnd = lastNewlineIndex + 2;
+                }, 0);
```

Comment:
There area several places use `setTimeout` with 0 delay. either we need to cleanup those with `useEffect` return or use `queueMicrotask` to defer the execution. Otherwise these can lead to memory leaks.